packages/angular/projects/ui-angular/src/lib/components/authenticator/components/setup-totp/setup-totp.component.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { AuthenticatorService } from '../../../../services/authenticator.service';
+import { SetupTotpComponent } from './setup-totp.component';
+import { AmplifySlotComponent } from '../../../../utilities/amplify-slot/amplify-slot.component';
+import { ErrorComponent } from '../../../../primitives/error/error.component';
+import { BaseFormFieldsComponent } from '../base-form-fields/base-form-fields.component';
+import { FormFieldComponent } from '../form-field/form-field.component';
+import { ButtonComponent } from '../../../../primitives/button/button.component';
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import QRCode from 'qrcode';
+import { Auth } from 'aws-amplify';
+import { MockComponent } from 'ng-mocks';
+import { getTotpCode } from '@aws-amplify/ui';
+
+const mockUser = { username: 'username' };
+const mockContext = {
+  country_code: '100',
+  formFields: { setupTOTP: { QR: null } },
+  user: mockUser,
+};
+
+const DEFAULT_TOTP_ISSUER = 'AWSCognito';
+const SECRET_KEY = 'secretKey';
+
+const setupTOTPSpy = jest.spyOn(Auth, 'setupTOTP');
+
+const toDataURLSpy = jest.spyOn(QRCode, 'toDataURL');
+
+describe('SetupTotpComponent', () => {
+  let fixture: ComponentFixture<SetupTotpComponent>;
+  let component: SetupTotpComponent;
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+
+    setupTOTPSpy.mockResolvedValue(SECRET_KEY);
+
+    const mockAuthenticatorService = {
+      _state: {},
+      isPending: false,
+      authState: {
+        context: {
+          actorRef: { getSnapshot: () => ({ context: { ...mockContext } }) },
+        },
+      },
+      updateForm: jest.fn(),
+      submitForm: jest.fn(),
+      context: jest.fn().mockReturnValue({}),
+      slotContext: jest.fn().mockReturnValue({}),
```

Comment:
```suggestion
      context: jest.mockReturnValue({}),
      slotContext: jest.mockReturnValue({}),
```

Reply:
Maybe this is allowed in jest 28, but jest 27, it's throwing me on `mockReturnValue` does not exist on typeof jest.

Reply:
I had to use jest 27^ because `jest-preset-angular` is version 10 which supports Angular 11, and wasn't working in my tests with jest 28.